src/main/java/org/medicmobile/webapp/mobile/SettingsStore.java

```diff
@@ -19,9 +21,24 @@
 	}
 
 	public abstract String getAppUrl();
+
+	public String getRootUrl() {
+		String appUrl = getAppUrl();
+		return appUrl + (DISABLE_APP_URL_VALIDATION ? "" : "/medic/_design/medic/_rewrite/");
+	}
+
+	public String getUrlToLoad(Uri url) {
+		return url != null ? url.toString() : getRootUrl();
+	}
+
+	public boolean isRootUrl(String url) {
+		return getRootUrl().equals(url);
+	}
+
```

Comment:
This feels like a change from a separate commit - does this branch need a rebase, or am I missing something?

Reply:
I moved these functions from the `EmbededBrowserActivity` to the `SettingsStore` since it's related to the [appURL](https://github.com/medic/cht-android/pull/236/files#diff-d9bfe28a0aee5b476da7e9b261a3636471bf81c0c1a205d6aafb12804abdfe8bR23) and we use them in the [UrlHandler](https://github.com/medic/cht-android/pull/236/files#diff-6fc2d35d93c4d6e27de9ef66977f756e69232ecc16df384ae59ab7f5cf02fb64R51) and [EmbededActivity](https://github.com/medic/cht-android/pull/236/files#diff-367a8128fbed2e243594107b6a9d04bc87d7c1b3d79982ba7a660bf75ea63839R344).